### PR TITLE
Multiple NFs of the same type

### PR DESCRIPTION
**Pull request by jamiedparsons**
This PR adds an option to create an NSD where the operator can specify how many copies of an NF they want to deploy and the configuration for each of them.

I tested and unfortunately there is a bug that mean that this doesn't work for CNFs: https://portal.microsofticm.com/imp/v3/incidents/details/405078667/home

This PR shouldn't be updated until corresponding changes are made to the quickstarts.

### CODE REVIEW

File: src/aosm/azext_aosm/tests/latest/test_nsd.py [hunk 1]
```diff
@@ -6,16 +6,149 @@
...
+import shutil
+import subprocess
 from pathlib import Path
 from unittest.mock import patch
 from tempfile import TemporaryDirectory
-from typing import Any, Dict
+
+import jsonschema
 
 from azext_aosm.custom import generate_design_config, build_design
```
Comment (Cyclam) [inline src/aosm/azext_aosm/tests/latest/test_nsd.py:18]: I should have asked in the unit test review, but why aren't we using PyTest?

**Review by Cyclam — APPROVED** [review]
Some minor comments, but LGTM.

File: src/aosm/azext_aosm/tests/latest/metaschema.json [hunk 2]
```diff
@@ -0,0 +1,264 @@
+{
```
[the remark this answers is not in the document]
Comment (jamiedparsons) [reply src/aosm/azext_aosm/tests/latest/metaschema.json:1]: Yes - unfortunately you aren't allowed to add comments to JSON schemas, so I can't explain there.

This file isn't used anywhere, but I thought it was useful to leave in for reference.

File: src/aosm/azext_aosm/tests/latest/test_nsd.py [hunk 3]
```diff
@@ -6,16 +6,149 @@
...
+
+
+def validate_schema_against_metaschema(schema_data):
+    """
+    Validate that the schema produced by the CLI matches the AOSM metaschema.
+    """
+
+    # There is a bug in the jsonschema module that means that it hits an error in with
+    # the "$id" bit of the metaschema.  Here we use a modified version of the metaschema
+    # with that small section removed.
```
[the remark this answers is not in the document]
Comment (jamiedparsons) [reply src/aosm/azext_aosm/tests/latest/test_nsd.py:111]: Not sure what you mean by acceptable.  Jsonschema is a 3rd party library, I guess we could go and fix it, but it doesn't seem worth the bother as it is only used for our UTs.

File: src/aosm/azext_aosm/tests/latest/test_nsd.py (hunk 1 above)
Comment (jamiedparsons) [reply src/aosm/azext_aosm/tests/latest/test_nsd.py:18]: I think the basic examples for the CLI just used unit test, I was worried that `azdev test` would be incompatible with them.  But it looks like other extensions do use pytest, so we could move over.  I'm not going to do that right now though.

File: src/aosm/azext_aosm/generate_nsd/nsd_generator.py [hunk 4]
```diff
@@ -4,7 +4,6 @@
 # --------------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (jamiedparsons) [reply src/aosm/azext_aosm/generate_nsd/nsd_generator.py:4]: That's fine unless you think that there are some places where comments should be added.

**Pull request merged**